### PR TITLE
Rails 7 compatiblity and rebranding

### PR DESCRIPTION
**Pull request by avonderluft**
- update code, config, and tests for Rails 7
- add github actions workflow for CI
- Rails 5 no longer supported
- rubocop config updated, some linting
- add markdown option to admin/snippet
- add image, audio, and navigation tags
- ensure page reorder functions correctly
- Gem build tested locally, successfully ran on a new project

### CODE REVIEW

**Review by Judahmeek — APPROVED** [review]
Only found minor issues. Nothing blocking.

**Comment by justin808** [comment]
@avonderluft and @Judahmeek please merge when you guys are ready.

**Comment by avonderluft** [comment]
@justin808 - I'm going take a little time to go over @Judahmeek 's comments, and likely make adjustments accordingly before I merge.  On version number, I was thinking 3.0.0 to show a new release, but also  indicate continuity with ComfyMex, whose last release is 2.0.19.  If you'd rather go with 1.0.0 let me know before I merge.

File: README.md [hunk 1]
```diff
@@ -25,12 +19,17 @@ ComfortableMexicanSofa is a powerful Ruby on Rails 5.2+ CMS (Content Management
...
+- Ruby >= 3.0 with Rails >= 6.1
+- On Ruby 3.x, Rails 7.x + is recommended, since performance is noticably better than on 6.x
+
 ## Installation
 
 Add gem definition to your Gemfile:
 
 ```ruby
-gem "comfortable_mexican_sofa", "~> 2.0.0"
+gem "comfortable_media_surfer", "~> 3.0.0"
```
[the remark this answers is not in the document]
Comment (avonderluft) [reply README.md:32]: I'm leaving that decision to @justin808

File: README.md [hunk 2]
```diff
@@ -50,7 +49,7 @@ comfy_route :cms, path: "/"
 
 After finishing installation you should be able to navigate to http://localhost:3000/admin
 
-Default username and password is 'username' and 'password'. You probably want to change it right away. Admin credentials (among other things) can be found and changed in the cms initializer: [/config/initializers/comfortable\_mexican\_sofa.rb](https://github.com/comfy/comfortable-mexican-sofa/blob/master/config/initializers/comfortable_mexican_sofa.rb)
+Default username and password is 'username' and 'password'. You probably want to change it right away. Admin credentials (among other things) can be found and changed in the cms initializer: [/config/initializers/comfortable\_mexican\_sofa.rb](https://github.com/comfy/comfortable-mexican-sofa/blob/master/config/initializers/comfortable_media_surfer.rb)
```
[the remark this answers is not in the document]
Comment (avonderluft) [reply README.md:52]: fixed

File: app/controllers/concerns/comfy/paginate.rb [hunk 3]
```diff
@@ -1,16 +1,11 @@
 # frozen_string_literal: true
 
 module Comfy::Paginate
-
-  # Wrapper to deal with WillPaginate vs Kaminari nonsense
   def comfy_paginate(scope, per_page: 50)
-    if defined?(WillPaginate)
-      scope.paginate(page: params[:page], per_page: per_page)
```
[the remark this answers is not in the document]
Comment (avonderluft) [reply app/controllers/concerns/comfy/paginate.rb:8]: @Judahmeek - it had to do with Rails 7 compatibility, and I was being lazy.  We can re-add in a future release if there's a demand for it.

File: lib/generators/comfy/cms/README [hunk 4]
```diff
@@ -3,15 +3,26 @@
...
+
+Then go to http://your-rails-app/admin to start adding content.
+
+Default username and password are: user / pass
+Change these defaults in config/initializers/occams.rb
+
+----------------------------
+IF YOU ARE ON RAILS 6.x
+----------------------------
+and encounter errors referencing webpacker:
```
[the remark this answers is not in the document]
Comment (avonderluft) [reply lib/generators/comfy/cms/README:25]: I'll leave that to @justin808   I think in general we're moving on from Rails 6 and webpacker anyway.

File: test/helpers/admin_cms_helper_test.rb [hunk 5]
```diff
@@ -30,52 +28,51 @@ def test_comfy_form_with
...
+      attachment: @file.attachment,
+      multiple: true
     )
     assert_equal '{{ cms:page_file_link test, filename: "default.jpg", as: image }}', actual
   end
 
   def test_cms_file_link_tag
     actual = cms_file_link_tag(@file)
-    assert_equal "{{ cms:file_link 593363170, as: image }}", actual
+    assert_equal '{{ cms:image default file }}', actual
```
[the remark this answers is not in the document]
Comment (avonderluft) [reply test/helpers/admin_cms_helper_test.rb:70]: I don't remember details, except that this was to fix failing tests with Rails 7

File: test/lib/seeds/snippets_test.rb [hunk 6]
```diff
@@ -13,72 +12,68 @@ def test_creation
...
+    assert_equal 'Default Seed Snippet', @snippet.label
     assert_equal "Default Seed Snippet Content\n", @snippet.content
   end
 
   def test_delete
     old_snippet = @snippet
-    old_snippet.update_column(:identifier, "old")
+    old_snippet.update_column(:identifier, 'old')
 
-    assert_no_difference -> { Comfy::Cms::Snippet.count } do
```
[the remark this answers is not in the document]
Comment (avonderluft) [reply test/lib/seeds/snippets_test.rb:48]: same.  Fixing failing tests with Rails 7

**Pull request merged**